### ISSUE TITLE
potential for black-holing log messages via transitive dependency on gulplog

### ISSUE THREAD

**Issue by ischneider** [gulpjs/gulp-util#116]
`gulp-util.log` checks for `hasGulplog` on module initialization and if present, defers logging to `gulplog.info`, which in turn will eat any messages below `error` (I believe) unless there is explicitly an event handler for the other levels (info, warn, etc.).

This can happen 'accidentally' if a transitive dependency uses gulplog.

It _seems_ like the reasonable thing to do is to use `gulplog.error` to ensure all logging is output properly.

**Comment by phated** [comment]
`gulplog.info` is the default log level for gulp-cli, which is why we used it (aligns with `console.log`).  The only messages that aren't logged by default is `gulplog.debug`.  However, I think you might actually be saying that some dependency is using `gulplog`, which makes it available in the global scope but you aren't using the new gulp-cli, so no logs are being logged.  Is that correct?

**Comment by ischneider** [comment]
> However, I think you might actually be saying that some dependency is using gulplog, which makes it available in the global scope but you aren't using the new gulp-cli, so no logs are being logged. Is that correct?

Correct and I can verify that using the `gulp` executable from `gulp-cli` produces the desired behavior, i.e. `gulp-util.log` produces output even when `gulplog` is present.

This is fine longer term, but it seems like a potential problem for the unaware - case in point: me and my time lost :(

Super simple example
```
require('gulplog');
require('gulp-util').log('foobar');
```

When run using global gulp, no 'foobar'. When using locally installed `gulp` from `gulp-cli`, 'foobar' is logged.

Thanks :smile:

**Comment by bdukes** [comment]
Is the solution for `gulp-util.log` to check (the yet-to-be created) `hasGulpCli`, instead of `hasGulplog`?

**Comment by ischneider** [comment]
> Is the solution for gulp-util.log to check (the yet-to-be-created) hasGulpCli, instead of hasGulplog?

It could be? Unfortunately, I've moved on and no longer use gulp in my day-to-day work.

**Comment by demurgos** [comment]
I'd like to close this issue following the deprecation of `plugin-error`.
Is this problem still present? If so, where should this be moved, `gulplog`?

/cc @phated

**Comment by bdukes** [comment]
Yes, this is still a `gulplog` issue, I've created an SSCCE at https://gist.github.com/bdukes/5d3d48321ba96f035311ee9f63aace70

**Comment by phated** [comment]
This problem won't be solved until we get the community transferred off gulp installed globally ☹️
